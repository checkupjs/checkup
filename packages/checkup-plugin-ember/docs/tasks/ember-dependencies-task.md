<!--TASK_NAME_START-->
# ember/ember-dependencies
<!--TASK_NAME_END-->

<!--TASK_DESCRIPTION_START-->
Finds Ember-specific dependencies and their versions in an Ember.js project
<!--TASK_DESCRIPTION_END-->

<!--RUN_START-->
## To run this task

```bash
checkup run --task ember/ember-dependencies
```
<!--RUN_END-->
