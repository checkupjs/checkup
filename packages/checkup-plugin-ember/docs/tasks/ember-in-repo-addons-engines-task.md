<!--TASK_NAME_START-->
# ember/ember-in-repo-addons-engines
<!--TASK_NAME_END-->

<!--TASK_DESCRIPTION_START-->
Finds all in-repo engines and addons in an Ember.js project
<!--TASK_DESCRIPTION_END-->

<!--RUN_START-->
## To run this task

```bash
checkup run --task ember/ember-in-repo-addons-engines
```
<!--RUN_END-->
