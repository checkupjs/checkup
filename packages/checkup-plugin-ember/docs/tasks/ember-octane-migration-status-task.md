<!--TASK_NAME_START-->
# ember/ember-octane-migration-status
<!--TASK_NAME_END-->

<!--TASK_DESCRIPTION_START-->
Tracks the migration status when moving from Ember Classic to Ember Octane in an Ember.js project
<!--TASK_DESCRIPTION_END-->

<!--RUN_START-->
## To run this task

```bash
checkup run --task ember/ember-octane-migration-status
```
<!--RUN_END-->
