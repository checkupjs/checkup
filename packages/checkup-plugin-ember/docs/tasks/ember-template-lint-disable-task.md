<!--TASK_NAME_START-->
# ember/ember-template-lint-disables
<!--TASK_NAME_END-->

<!--TASK_DESCRIPTION_START-->
Finds all disabled ember-template-lint rules in an Ember.js project
<!--TASK_DESCRIPTION_END-->

<!--RUN_START-->
## To run this task

```bash
checkup run --task ember/ember-template-lint-disables
```
<!--RUN_END-->
