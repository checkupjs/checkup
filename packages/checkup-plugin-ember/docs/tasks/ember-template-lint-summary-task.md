<!--TASK_NAME_START-->
# ember/ember-template-lint-summary
<!--TASK_NAME_END-->

<!--TASK_DESCRIPTION_START-->
Gets a summary of all ember-template-lint results in an Ember.js project
<!--TASK_DESCRIPTION_END-->

<!--RUN_START-->
## To run this task

```bash
checkup run --task ember/ember-template-lint-summary
```
<!--RUN_END-->
