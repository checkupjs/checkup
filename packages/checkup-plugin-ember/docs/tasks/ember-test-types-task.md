<!--TASK_NAME_START-->
# ember/ember-test-types
<!--TASK_NAME_END-->

<!--TASK_DESCRIPTION_START-->
Gets a breakdown of all test types in an Ember.js project
<!--TASK_DESCRIPTION_END-->

<!--RUN_START-->
## To run this task

```bash
checkup run --task ember/ember-test-types
```
<!--RUN_END-->
