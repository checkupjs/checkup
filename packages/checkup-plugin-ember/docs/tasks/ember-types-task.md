<!--TASK_NAME_START-->
# ember/ember-types
<!--TASK_NAME_END-->

<!--TASK_DESCRIPTION_START-->
Gets a breakdown of all Ember types in an Ember.js project
<!--TASK_DESCRIPTION_END-->

<!--RUN_START-->
## To run this task

```bash
checkup run --task ember/ember-types
```
<!--RUN_END-->
