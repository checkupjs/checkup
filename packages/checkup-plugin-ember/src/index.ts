import { RegistrationArgs, getPluginName } from '@checkup/core';

import { evaluateActions as evaluateTemplateLintDisables } from './actions/ember-template-lint-disable-actions';
import { evaluateActions as evaluateTemplateLintSummary } from './actions/ember-template-lint-summary-actions';
import { evaluateActions as evaluateTestTypes } from './actions/ember-test-types-actions';
import { report as reportEmberTestTypes } from './reporters/ember-test-types-reporter';
import { report as reportEmberOctaneMigrationStatus } from './reporters/ember-octane-migration-status-reporter';
import { report as reportEmberTemplateLintSummary } from './reporters/ember-template-lint-summary-reporter';
import EmberDependenciesTask from './tasks/ember-dependencies-task';
import EmberInRepoAddonsEnginesTask from './tasks/ember-in-repo-addons-engines-task';
import EmberTestTypesTaskTask from './tasks/ember-test-types-task';
import EmberTypesTask from './tasks/ember-types-task';
import EmberTemplateLintDisableTask from './tasks/ember-template-lint-disable-task';
import EmberTemplateLintSummaryTask from './tasks/ember-template-lint-summary-task';
import EmberOctaneMigrationStatusTask from './tasks/ember-octane-migration-status-task';

export function register(args: RegistrationArgs) {
  let pluginName = getPluginName(__dirname);

  args.register.actions('ember-template-lint-disables', evaluateTemplateLintDisables);
  args.register.actions('ember-template-lint-summary', evaluateTemplateLintSummary);
  args.register.actions('ember-test-types', evaluateTestTypes);

  args.register.taskReporter('ember-test-types', reportEmberTestTypes);
  args.register.taskReporter('ember-octane-migration-status', reportEmberOctaneMigrationStatus);
  args.register.taskReporter('ember-template-lint-summary', reportEmberTemplateLintSummary);

  args.register.task(new EmberTypesTask(pluginName, args.context));
  args.register.task(new EmberDependenciesTask(pluginName, args.context));
  args.register.task(new EmberInRepoAddonsEnginesTask(pluginName, args.context));
  args.register.task(new EmberTestTypesTaskTask(pluginName, args.context));
  args.register.task(new EmberTemplateLintDisableTask(pluginName, args.context));
  args.register.task(new EmberTemplateLintSummaryTask(pluginName, args.context));
  args.register.task(new EmberOctaneMigrationStatusTask(pluginName, args.context));
}
