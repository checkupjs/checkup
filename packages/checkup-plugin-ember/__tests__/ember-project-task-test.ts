import { EmberProject, stdout } from '@checkup/test-helpers';
import { EmberProjectTask } from '../src/tasks';
import { EmberProjectTaskResult } from '../src/results';

describe('project-info-task', () => {
  let emberProject: EmberProject;

  describe('for Ember Applications', () => {
    beforeEach(() => {
      let packageJson = require('./__fixtures__/app-package.json');

      emberProject = new EmberProject('checkup-app', '0.0.0', (project) => {
        project.addDependency('ember-cli', '^3.15.0');
      });

      emberProject.updatePackageJson(packageJson);

      emberProject.writeSync();
    });

    afterEach(() => {
      emberProject.dispose();
    });

    it('can read project info and output to console', async () => {
      const result = await new EmberProjectTask({ path: emberProject.baseDir }).run();
      const taskResult = <EmberProjectTaskResult>result;

      taskResult.stdout();

      expect(stdout()).toMatchSnapshot();
    });

    it('can read project info as JSON', async () => {
      const result = await new EmberProjectTask({ path: emberProject.baseDir }).run();
      const taskResult = <EmberProjectTaskResult>result;

      expect(taskResult.json()).toMatchSnapshot();
    });

    it('can read project info as PDF', async () => {
      const result = await new EmberProjectTask({ path: emberProject.baseDir }).run();
      const taskResult = <EmberProjectTaskResult>result;

      expect(taskResult.pdf()).toMatchSnapshot();
    });
  });

  describe('for Ember Addons', () => {
    beforeEach(() => {
      let packageJson = require('./__fixtures__/addon-package.json');

      emberProject = new EmberProject('checkup-app', '0.0.0', (project) => {
        project.addDependency('ember-cli', '^3.15.0');
      });

      emberProject.updatePackageJson(packageJson);

      emberProject.writeSync();
    });

    afterEach(() => {
      emberProject.dispose();
    });

    it('can read project info and output to console', async () => {
      const result = await new EmberProjectTask({ path: emberProject.baseDir }).run();
      const taskResult = <EmberProjectTaskResult>result;

      taskResult.stdout();

      expect(stdout()).toMatchSnapshot();
    });

    it('can read project info as JSON', async () => {
      const result = await new EmberProjectTask({ path: emberProject.baseDir }).run();
      const taskResult = <EmberProjectTaskResult>result;

      expect(taskResult.json()).toMatchSnapshot();
    });

    it('can read project info as PDF', async () => {
      const result = await new EmberProjectTask({ path: emberProject.baseDir }).run();
      const taskResult = <EmberProjectTaskResult>result;

      expect(taskResult.pdf()).toMatchSnapshot();
    });
  });
});
