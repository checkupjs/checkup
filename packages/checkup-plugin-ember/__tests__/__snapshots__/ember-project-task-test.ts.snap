// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project-info-task for Ember Addons can read project info and output to console 1`] = `
"=== Ember Project

type: Ember.js addon

"
`;

exports[`project-info-task for Ember Addons can read project info as JSON 1`] = `
Object {
  "meta": Object {
    "friendlyTaskName": "Ember Project",
    "taskClassification": Object {
      "category": "recommendations",
      "priority": "medium",
    },
    "taskName": "ember-project",
  },
  "result": Object {
    "type": "Ember.js addon",
  },
}
`;

exports[`project-info-task for Ember Addons can read project info for html 1`] = `
Array [
  TableData {
    "componentType": "table",
    "formattedResult": Array [
      Object {
        "result": Object {
          "name": "type",
          "value": "Ember.js addon",
        },
        "rowClass": "bg-gray-100",
      },
    ],
    "meta": Object {
      "friendlyTaskName": "Ember Project",
      "taskClassification": Object {
        "category": "recommendations",
        "priority": "medium",
      },
      "taskName": "ember-project",
    },
    "tableHeaders": Array [
      "name",
      "value",
    ],
  },
]
`;

exports[`project-info-task for Ember Applications can read project info and output to console 1`] = `
"=== Ember Project

type: Ember.js application

"
`;

exports[`project-info-task for Ember Applications can read project info as JSON 1`] = `
Object {
  "meta": Object {
    "friendlyTaskName": "Ember Project",
    "taskClassification": Object {
      "category": "recommendations",
      "priority": "medium",
    },
    "taskName": "ember-project",
  },
  "result": Object {
    "type": "Ember.js application",
  },
}
`;

exports[`project-info-task for Ember Applications can read project info for html 1`] = `
Array [
  TableData {
    "componentType": "table",
    "formattedResult": Array [
      Object {
        "result": Object {
          "name": "type",
          "value": "Ember.js application",
        },
        "rowClass": "bg-gray-100",
      },
    ],
    "meta": Object {
      "friendlyTaskName": "Ember Project",
      "taskClassification": Object {
        "category": "recommendations",
        "priority": "medium",
      },
      "taskName": "ember-project",
    },
    "tableHeaders": Array [
      "name",
      "value",
    ],
  },
]
`;
